twyla/service/queues.py

```diff
@@ -49,6 +50,25 @@ def __init__(self):
         await self.get_connection()
         self.channel = await self.protocol.channel()
         await self.declare()
+        return asyncio.ensure_future(self.cancel_on_disconnect())
+
+
+    async def cancel_on_disconnect(self):
+        await self.protocol.wait_closed()
+        await self.stop()
+        for task in asyncio.Task.all_tasks():
+            # Cancel all pending tasks (this should be only the current method
+            # and the event listener in most cases). Make sure to not cancel
+            # this method.
+            my_class_name = self.__class__.__name__
```

Comment:
Could this maybe be explained a bit? This is what (I think) I understand that it cancels any remaining tasks when a queue disconnection happens, and this function can "know" when to do that because it's given the signal somewhere that the queue has been disconnected.

Reply:
Yes, will add a comment.
Basically `await self.protocol.wait_closed()` will block this coroutine until the connection to rabbit closes for any reason. Then the rest of the function is executed cleaning the up all the things and raising the `CancelledError`.

Reply:
It could instead of raising just call `self.connect()` again but I think it makes more sense to leave that decision to the user of the library.